pytest_django/fixtures.py

```diff
@@ -174,7 +176,19 @@ class PytestDjangoTestCase(test_case_class):  # type: ignore[misc,valid-type]
         if _databases is not None:
             databases = _databases
 
+        @classmethod
+        def django_tear_down_class(cls) -> None:
+            cls._remove_databases_failures()
+            if hasattr(cls, '_cls_modified_context'):
+                cls._cls_modified_context.disable()
+                delattr(cls, '_cls_modified_context')
+            if hasattr(cls, '_cls_overridden_context'):
+                cls._cls_overridden_context.disable()
+                delattr(cls, '_cls_overridden_context')
+
     PytestDjangoTestCase.setUpClass()
+    if VERSION >= (4, 0):
+        request.addfinalizer(PytestDjangoTestCase.django_tear_down_class)
```

Comment:
Does the following work, instead of copying `django_tear_down_class` over?

```suggestion
        request.addfinalizer(PytestDjangoTestCase.doClassCleanups)
```

Reply:
Yes, perfect!